src/test/java/de/tum/in/www1/artemis/assessment/ResultServiceIntegrationTest.java

```diff
@@ -722,4 +724,40 @@ void testGetAssessmentCountByCorrectionRoundForProgrammingExercise() {
         assertThat(assessments[0].inTime()).isEqualTo(1);    // correction round 1
         assertThat(assessments[1].inTime()).isEqualTo(1);    // correction round 2
     }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testGetAllFeedbackDetailsForExercise() throws Exception {
+        ProgrammingExercise programmingExercise = programmingExerciseUtilService.addProgrammingExerciseToCourse(course);
+        StudentParticipation participation = participationUtilService.createAndSaveParticipationForExercise(programmingExercise, TEST_PREFIX + "student1");
+        Result result = participationUtilService.addResultToParticipation(AssessmentType.AUTOMATIC, null, participation);
+        ProgrammingExerciseTestCase testCase = programmingExerciseUtilService.addTestCaseToProgrammingExercise(programmingExercise, "test1");
+        testCase.setId(1L);
+
+        Feedback feedback = new Feedback();
+        feedback.setPositive(false);
+        feedback.setDetailText("Some feedback");
+        feedback.setTestCase(testCase);
+        participationUtilService.addFeedbackToResult(feedback, result);
+
+        List<FeedbackDetailDTO> response = request.getList("/api/exercises/" + programmingExercise.getId() + "/feedback-details", HttpStatus.OK, FeedbackDetailDTO.class);
+
+        assertThat(response).isNotEmpty();
+        FeedbackDetailDTO feedbackDetail = response.getFirst();
+        assertThat(feedbackDetail.count()).isEqualTo(1);
+        assertThat(feedbackDetail.relativeCount()).isEqualTo(100.0);
+        assertThat(feedbackDetail.detailText()).isEqualTo("Some feedback");
+        assertThat(feedbackDetail.testCaseName()).isEqualTo("test1");
+        assertThat(feedbackDetail.taskNumber()).isEqualTo(1);
+    }
```

Comment:
I think at least one further test case that checks the aggregation and count would be beneficial.
E.g. the feedback detail has at least the count 2 and at least 2 different messages